roundtripper.go

```diff
@@ -100,7 +100,7 @@ func (rt *roundTripper) getTransport(req *http.Request, addr string) error {
 		return fmt.Errorf("invalid URL scheme: [%v]", req.URL.Scheme)
 	}
 
-	_, err := rt.dialTLS(context.Background(), "tcp", addr)
```

Comment:
what's this change for?

Reply:
It's just that now we are losing the context between NewRequestWithContext and the request. I wrote next message with example